libhoney/transmission.py

```diff
@@ -92,23 +97,29 @@ def _send(self, ev):
             "body": resp.text,
             "error": "",
         }
-        try:
-            if self.block_on_response:
-                self.responses.put(response)
-            else:
+        if self.block_on_response:
+            self.responses.put(response)
+        else:
+            try:
                 self.responses.put_nowait(response)
-        except queue.Full:
-            pass
+            except queue.Full:
+                pass
 
     def close(self):
         '''call close to send all in-flight requests and shut down the
            senders nicely'''
         for i in range(self.max_concurrent_batches):
-            self.pending.put(None)
+            try:
+                self.pending.put(None, true, 10)
+            except queue.Full:
+                pass
         for t in self.threads:
             t.join()
         # signal to the responses queue that nothing more is coming.
-        self.responses.put(None)
+        try:
+            self.responses.put(None, true, 10)
```

Comment:
yeah, since it's on close, that says "wait 10 seconds to try and send indicate that the queue is done, and if, in that time, you couldn't add this event, just carry on."  https://docs.python.org/2/library/queue.html#Queue.Queue.put

This is on the theory that if honeycomb is really backed up and the queue isn't draining at all we shouldn't hang out forever trying to close the library.